Convert texture and settings usages to thinhashes

ryanjsims opened the pull request:
These have always been thin hashes with extra steps, and it'll be less work to add new settings usages or texture usages if they're treated as regular thin hashes

Additionally I've pulled in the current work I've done on adding DXBC parsing so that I can use it to retrieve the settings and texture names from the shaders with the material-resource-name-dumper tool.

Finally this also has a couple of fixes to the illuminate material parsing since AH apparently changed how the illuminate shader works recently

ryanjsims committed:
Remove settings_usage and texture_usage types and replace with direct thin hash use, adding the texture usages and settings usages as strings to thinhashes.txt

ryanjsims committed:
Add material-resource-name-dumper tool to export texture names and settings names from materials

xypwn commented on stingray/unit/material/d3d/dxbc.go:
```diff
@@ -0,0 +1,132 @@
+}
+
+func (d *DXBC) ToGLSL() string {
+	toReturn := "#version 420 core\n\n"
```
Looks like this means we'll (eventually) have to upgrade to OpenGL 4.2 (using 3.2 at the moment and GLSL 3.3).
Or do you think the feature set we need would be supported in OpenGL 3.x?

Even if we were forced to use 4.2, I think it should be fine in any case, since any hardware capable of running HD2 will definitely support GL 4.2.

xypwn commented on extractor/material/extractor.go:
```diff
@@ -341,28 +377,28 @@ func compareMaterials(doc *gltf.Document, mat *material.Material, matIdx uint32,
-			switch usage {
-			case MaterialLUT:
+			switch texUsage {
+			case tex_material_lut:
```
I was thinking instead of declaring `tex_XXX`, since all these hashes are known anyways, we can just use `texUsageStr, ok := ctx.ThinHashes()[texUsage]` and then `switch texUsageStr { case "material_lut": ... }`.

Side note: For auto-generating enum strings, you can use https://pkg.go.dev/golang.org/x/tools/cmd/stringer with `go generate`.

xypwn reviewed (commented):
Really nice work! Thanks for the fixes, and I think the dxbc analyzer/translator is looking great so far!

ryanjsims committed:
Use strings directly and revert illuminate material fix in this branch

Since it would just make more work when rebasing the visibility mask branch onto these changes, I removed the illuminate material fix from this branch
